File: lib_book_common/src/main/java/com/ebook/common/util/SoftInputUtil.java
package com.ebook.common.util;

import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

/**
 * Description: <软键盘的显示与隐藏><br>
 * <ul>
 * <li>1.显示软键盘</li>
 * <li>2.隐藏软键盘</li>
 * </ul>
 */
public class SoftInputUtil {

    /**
     * 显示软键盘
     *
     * @param context
     */
    public static void showSoftInput(Context context) {
        InputMethodManager imm =
                (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE); // 显示软键盘
        imm.toggleSoftInput(0, InputMethodManager.HIDE_NOT_ALWAYS);
    }

    /**
     * 显示软键盘
     *
     * @param context
     */
    public static void showSoftInput(Context context, View view) {
        InputMethodManager imm =
                (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE); // 显示软键盘
        imm.showSoftInput(view, 0);
    }

    /**
     * 隐藏软键盘
     *
     * @param context
     * @param view
     */
    public static void hideSoftInput(Context context, View view) {
        InputMethodManager immHide =
                (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE); // 隐藏软键盘
        immHide.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }
}

File: lib_book_common/src/main/java/com/ebook/common/util/SoftInputUtil.kt
package com.ebook.common.util

import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager

/**
 * Description: <软键盘的显示与隐藏>
 *
 *  * 1.显示软键盘
 *  * 2.隐藏软键盘
 *
 */
object SoftInputUtil {
    /**
     * 显示软键盘
     */
    @JvmStatic
    fun showSoftInput(context: Context, view: View) {
        val imm = context.getSystemService(
            InputMethodManager::class.java
        ) // 显示软键盘
        imm.showSoftInput(view, 0)
    }

    /**
     * 隐藏软键盘
     */
    @JvmStatic
    fun hideSoftInput(context: Context, view: View) {
        val immHide = context.getSystemService(
            InputMethodManager::class.java
        ) // 隐藏软键盘
        immHide.hideSoftInputFromWindow(view.windowToken, 0)
    }
}